bin/ch/Helpers.cpp

```diff
@@ -4,6 +4,212 @@
 //-------------------------------------------------------------------------------------------------------
 #include "stdafx.h"
 
+//TODO: x-plat definitions
+#ifdef _WIN32
+typedef char16 TTDHostCharType;
+typedef struct _wfinddata_t TTDHostFileInfo;
+typedef intptr_t TTDHostFindHandle;
+typedef struct _stat TTDHostStatType;
+
+#define TTDHostPathSeparator _u("\\")
+#define TTDHostPathSeparatorChar _u('\\')
+#define TTDHostFindInvalid -1
+
+size_t TTDHostStringLength(const TTDHostCharType* str)
+{
+    return wcslen(str);
+}
+
+void TTDHostInitEmpty(TTDHostCharType* dst)
+{
+    dst[0] = _u('\0');
+}
+
+void TTDHostInitFromUriBytes(TTDHostCharType* dst, const byte* uriBytes, size_t uriBytesLength)
```

Comment:
Why do you design "uri" this way... makes its usage on xplat very hard. Looks the "uri" string has to contain char16, and pass around as raw pointer, here you assemble it back to char16*.

Can it simply be type `const char*`? You can use conversion routines in Utf8Codex dir.